tests/unit/s2n_override_openssl_random_test.c

```diff
@@ -131,6 +131,9 @@ int main(int argc, char **argv)
     EXPECT_SUCCESS(s2n_stuffer_free(&dhparams_in));
     EXPECT_SUCCESS(s2n_stuffer_free(&test_entropy));
     free(dhparams_pem);
+    /* s2n_rand_set_callbacks overrode the default callbacks without cleaning up the default callbacks.
+     * Call defaults rand cleanup function to cleanup old callbacks before setting new ones. */
+    POSIX_GUARD_RESULT(s2n_rand_cleanup());
```

Comment:
why is this different from s2n_drbg_test.c?

Reply:
I tried to do the same as `s2n_drbg_test.c`, but it causes the test to fail.

The reason is `s2n_override_openssl_random_test.c` rans
https://github.com/aws/s2n-tls/blob/d640c7427e476a1bb794f7e695b063bd730f2ae7/tests/unit/s2n_override_openssl_random_test.c#L115
after it resets all callbacks. That will require the libcrypto engine. If we call `s2n_rand_cleanup()` before `s2n_rand_set_callbacks`, then the libcrypto engine will be shut down, so that `s2n_dh_generate_ephermeral_key` can't be ran properly. We clean up the libcrypto engine like this:
https://github.com/aws/s2n-tls/blob/d640c7427e476a1bb794f7e695b063bd730f2ae7/utils/s2n_random.c#L598-L610

I detected this issue by running gdb to track which callback was called before and after the change. I found out that the `s2n_entropy_generator` is called in `s2n_dh_generate_ephermeral_key` before the change, but it is not called after the change. I then comment out the part where we cleanup the engine, and the test didn't detect any open fds (but it did detect memory loss in engines which is expected).

Hence, I think it makes sense to leave `s2n_ran_cleanup` at the end with all the cleanup functions. Please let me know what you think.